host-bmc/utils.cpp

```diff
@@ -92,11 +107,24 @@ void addObjectPathEntityAssociations(
             {
                 continue;
             }
+            /*If chassis number is not empty, it means we've encountered
+              numbered chassis. So we prepend previous entity instance id and
+              chassis number to prepare unique leafnodes
+            */
+            if (!chassisNum.empty() &&
+                node_entity.entity_type != PLDM_ENTITY_SYSTEM_CHASSIS)
+            {
+                leafNode = (chassisNum + previousEntityInstanceNum) + "_" +
+                           entityNode;
```

Comment:
prefer `std::format` to format `arbitrary strings` rather than string appends manually.